Sources/Plasma/FeatureLib/pfPython/plPythonConvert.h

```diff
@@ -0,0 +1,205 @@
+    {
+        return PyUnicode_FromString(value);
+    }
+
+    inline PyObject* ConvertFrom(double value)
+    {
+        return PyFloat_FromDouble(value);
+    }
+
+    inline PyObject* ConvertFrom(float value)
+    {
+        return PyFloat_FromDouble(value);
+    }
+
+    inline PyObject* ConvertFrom(signed char value)
+    {
+        return PyLong_FromLong(value);
+    }
+
+    inline PyObject* ConvertFrom(signed short value)
+    {
+        return PyLong_FromLong(value);
+    }
+
+    inline PyObject* ConvertFrom(signed int value)
+    {
+        return PyLong_FromLong(value);
+    }
+
+    inline PyObject* ConvertFrom(signed long value)
+    {
+        return PyLong_FromLong(value);
+    }
+
+    inline PyObject* ConvertFrom(signed long long value)
+    {
+        return PyLong_FromLongLong(value);
+    }
+
+    inline PyObject* ConvertFrom(ST::string&& value)
+    {
+        return PyUnicode_FromSTString(value);
+    }
+
+    inline PyObject* ConvertFrom(unsigned char value)
+    {
+        return PyLong_FromUnsignedLong(value);
+    }
+
+    inline PyObject* ConvertFrom(unsigned short value)
+    {
+        return PyLong_FromUnsignedLong(value);
+    }
+
+    inline PyObject* ConvertFrom(unsigned int value)
+    {
+        return PyLong_FromUnsignedLong(value);
+    }
+
+    inline PyObject* ConvertFrom(unsigned long value)
+    {
+        return PyLong_FromUnsignedLong(value);
+    }
+
+    inline PyObject* ConvertFrom(unsigned long long value)
+    {
+        return PyLong_FromUnsignedLongLong(value);
+    }
+
+    inline PyObject* ConvertFrom(wchar_t value)
+    {
+        return PyUnicode_FromFormat("%c", (int)value);
+    }
+
+    inline PyObject* ConvertFrom(std::nullptr_t)
+    {
+        Py_RETURN_NONE;
+    }
+
+    /**
+     * Returns a stolen reference to match the other overloads.
+     */
+    inline PyObject* ConvertFrom(PyObject* value)
+    {
+        return value;
+    }
+
+    /**
+     * Returns a stolen reference to match the other overloads.
+     */
+    inline PyObject* ConvertFrom(pyObjectRef&& value)
+    {
+        return value.Release();
+    }
+
+    namespace _detail
+    {
+        template<size_t _IdxT, typename... _TupleArgsT>
+        inline void ConvertToTuple(PyObject* result, std::tuple<_TupleArgsT...>&& tuple)
+        {
+            using _TupleT = std::decay_t<decltype(tuple)>;
+
+            if constexpr (_IdxT < std::tuple_size_v<_TupleT>) {
+                using _ElementT = std::tuple_element_t<_IdxT, _TupleT>;
+                PyTuple_SET_ITEM(
+                    result,
+                    _IdxT,
+                    plPython::ConvertFrom(
+                        std::forward<_ElementT>(
+                            std::get<_IdxT>(tuple)
+                        )
+                    )
+                );
+                ConvertToTuple<_IdxT + 1, _TupleArgsT...>(result, std::forward<_TupleT>(tuple));
+            }
+        }
+    };
```

Comment:
I think I need some comments for this stuff. I don't expect that I'll understand it fully, but just... something.